### PR TITLE
Plug 1.0 Upgrade - Runtime Setup

### LINKED ISSUE #66: Plug 1.0 Upgrade - Runtime Setup
Prepare the CENNZnet node runtime definition for further 1.0 upgrade work.

Tasks:

-  [x] Remove `crml` and `prml` Modules from the node runtime definition.
These crates may be added back into the runtime definition when they are individually upgraded.

https://github.com/cennznet/cennznet/blob/ec299bfb0d5230ac2388ef021ee129205f8b8d4a/runtime/src/lib.rs#L251-L284

- [ ] The runtime extrinsic types should be changed to `sr_primitives::UncheckedExtrinsic` and `sr_primitives::CheckedExtrinsic` and `SignedExtra`payload added

https://github.com/plugblockchain/plug-blockchain/blob/2c18dfb984df5f4c42adfb8fc734fea91df6a38b/node/runtime/src/lib.rs#L543-L560

A/C - It compiles (Tests will be fixed on a per crate basis)

### PR DESCRIPTION
**Pull request by cowboy-bebug**
Used [plug/1.0.0-rc1/node](https://github.com/plugblockchain/plug-blockchain/tree/1.0.0-rc1/node) as reference, because `node-template` was too bare bones

Changes:
```
src/                              
├── chain_spec        => will need to be updated as we add more modules
│   ├── dev.rs          -> merged into a single `chain_spec.rs`
│   ├── mod.rs          -> merged into a single `chain_spec.rs`
│   └── testnet.rs      -> merged into a single `chain_spec.rs`
│
├── cli.rs            => refactored to use plug/node-template
├── error.rs          => removed as it's not used anywhere
├── main.rs           => refactored to use plug/node-template
└── service.rs        => refactored to use plug/node-template

runtime/
├── Cargo.toml        => refactored to use plug/node-template
│
├── src               => refactored to use plug/node-template
│   ├── doughnut.rs     -> removed (will be added back later)
│   ├── fee.rs          -> removed
│   └── lib.rs          -> refactored to use plug/node-template
│
├── tests             => removed tests as they aren't specific to CENNZnet
│   ├── codec.rs        -> removed
│   └── fee.rs          -> removed
│
└── wasm              => wasm is achieved via build.rs now
    ├── Cargo.lock      -> removed
    ├── Cargo.toml      -> removed
    ├── build.sh        -> removed
    └── src             -> removed
        └── lib.rs      -> removed
```

Closes #66

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Service and ServiceFactory implementation. Specialized wrapper over substrate service.
+
+use aura_primitives::sr25519::AuthorityPair as AuraPair;
```
Comment (jordy25519) [inline src/service.rs]: Will need to rectify this with `plug/node` to use babe instead

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -12,162 +12,235 @@
...
 // `construct_runtime!` does a lot of recursion and requires us to increase the limit to 256.
 #![recursion_limit = "256"]
 
-use authority_discovery_primitives::{AuthorityId as EncodedAuthorityId, Signature as EncodedSignature};
-use babe_primitives::{AuthorityId as BabeId, AuthoritySignature as BabeSignature};
+// Make the WASM binary available.
+#[cfg(feature = "std")]
+include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+use aura_primitives::sr25519::AuthorityId as AuraId;
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Must use `babe` instead of `aura`

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -12,162 +12,235 @@
...
-	// Per convention: if the runtime behavior changes, increment spec_version
-	// and set impl_version to equal spec_version. If only runtime
-	// implementation changes and behavior does not, then leave spec_version as
-	// is and increment impl_version.
-	spec_version: 182,
-	impl_version: 182,
+	spec_name: create_runtime_str!("cennznet"),
+	impl_name: create_runtime_str!("centrality-cennznet"),
+	authoring_version: 1,
+	spec_version: 25,
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Bump to `26`

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -177,347 +250,38 @@ impl balances::Trait for Runtime {
...
+		NodeBlock = opaque::Block,
 		UncheckedExtrinsic = UncheckedExtrinsic
 	{
 		System: system::{Module, Call, Storage, Config, Event},
-		Utility: utility::{Module, Call, Event},
-		Babe: babe::{Module, Call, Storage, Config, Inherent(Timestamp)},
 		Timestamp: timestamp::{Module, Call, Storage, Inherent},
-		Authorship: authorship::{Module, Call, Storage, Inherent},
-		Indices: indices,
+		Aura: aura::{Module, Config<T>, Inherent(Timestamp)},
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Babe

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -177,347 +250,38 @@ impl balances::Trait for Runtime {
...
+		Grandpa: grandpa::{Module, Call, Storage, Config, Event},
+		Indices: indices::{default, Config<T>},
 		Balances: balances::{default, Error},
 		TransactionPayment: transaction_payment::{Module, Storage},
-		Staking: staking::{default, OfflineWorker},
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Need to keep staking also

**Review by jordy25519 — CHANGES_REQUESTED** [review]
This is right on track 👍 
Main changes required
- Must use babe not aura (part of the reason for upgrade)
- Must keep staking module

File: runtime/src/lib.rs (hunk 3 above)
Comment (jordy25519) [reply runtime/src/lib.rs]: bump to `impl_version: 26` also 👌

File: Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	"executor",
+	"primitives",
+	# "rpc",
+	# "rpc-client",
+	"runtime",
+	"testing",
+]
+
+[badges]
+travis-ci = { repository = "cennznet/cennznet", branch = "develop" }
```
Comment (jordy25519) [inline Cargo.toml]: need to change to circleci

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -69,7 +71,7 @@ include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
 /// Runtime version.
 pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("node"),
-	impl_name: create_runtime_str!("substrate-node"),
+	impl_name: create_runtime_str!("plug-node"),
```
Comment (jordy25519) [inline runtime/src/lib.rs]: cennznet-node

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -1,25 +1,28 @@
...
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The Substrate runtime. This can be compiled with ``#[no_std]`, ready for Wasm.
```
Comment (jordy25519) [inline runtime/src/lib.rs]: The CENNZnet runtime.

File: cli/src/chain_spec.rs [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+#[derive(Default, Clone, Serialize, Deserialize, ChainSpecExtension)]
+pub struct Extensions {
+	/// Block numbers with known hashes.
+	pub fork_blocks: client::ForkBlocks<Block>,
+}
+
+/// Specialized `ChainSpec`.
+pub type ChainSpec = substrate_service::ChainSpec<GenesisConfig, Extensions>;
+/// Flaming Fir testnet generator
+pub fn flaming_fir_config() -> Result<ChainSpec, String> {
```
Comment (jordy25519) [inline cli/src/chain_spec.rs]: Can remove all references to `flaming_fir`, CENNZnet doesn't need it

File: Cargo.toml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+[build-dependencies]
+vergen = "3.0.4"
+
+[workspace]
+members = [
+	"cli",
+	"executor",
+	"primitives",
+	# "rpc",
```
Comment (jordy25519) [inline Cargo.toml]: Can add these crates in to this PR

**Review by jordy25519 — CHANGES_REQUESTED** [review]
GJ on this, needs some small changes

File: cli/src/chain_spec.rs (hunk 9 above)
Comment (jordy25519) [reply cli/src/chain_spec.rs]: I'll open a task to do this in Plug also

File: runtime/src/lib.rs (hunk 7 above)
Comment (cowboy-bebug) [reply runtime/src/lib.rs]: My bad, I'll change it to this:
```rust
pub const VERSION: RuntimeVersion = RuntimeVersion {
	spec_name: create_runtime_str!("cennznet"),
	impl_name: create_runtime_str!("centrality-cennznet"),
	authoring_version: 1,
	spec_version: 26, // was 25
	impl_version: 26,
	apis: RUNTIME_API_VERSIONS,
};
```

**Review by jordy25519 — CHANGES_REQUESTED** [review]
LGTM! 🏅 🏅 🏅

**Review by jordy25519 — APPROVED** [review]

**Pull request merged**